### PR TITLE
Topology Filters

### LINKED ISSUE #1116: stopped/system containers hard to spot in sea of containers
at the moment one can choose between
- showing system container or not
- showing stopped containers or not

However, if you have loads of running/non-system containers, then if you want to look at stopped/system containers they get lost in the sea of containers.

Suggestion: make the switches three-way:
- system - both - user (defaulting to 'user')
- stopped - both - running (defaulting to 'running')

### PR DESCRIPTION
**Pull request by davkal**
Fixes #1116 

* topology filters overhaul
* allow SHOW X, SHOW Y (which is NOT X), SHOW BOTH

### CODE REVIEW

File: client/app/scripts/stores/app-store.js [hunk 1]
```diff
@@ -80,13 +80,11 @@ function setTopology(topologyId) {
...
   topologyList.forEach(topology => {
     let defaultOptions = makeOrderedMap();
-    if (topology.has('options')) {
-      topology.get('options').forEach((items, option) => {
-        items.forEach(item => {
-          if (item.get('default') === true) {
-            defaultOptions = defaultOptions.set(option, item.get('value'));
-          }
-        });
+    if (topology.has('options') && topology.get('options')) {
```
Comment (foot) [inline client/app/scripts/stores/app-store.js]: simplified to just `topology.get('options')`

**Comment by foot** [comment]
I like the aesthetic!

- Preloading positions by selecting app-containers then system-containers, and _then_ both causes bad node overlap sometimes, can be force-layed-out though. I'm not sure if there's an easy fix here? I guess you could include filtered nodes in the layout and then hide, though this could lead to sub-optimal layouts...
- Maybe give the unselected items a bg fill as they are already at 0.7opacity, looks quite nice. Otherwise a bit messy in the rare case of node/control overlap down there.
- I quite like having the two bars the same length, but it sounds like that option was tried and outvoted. :)
- Even when "both" is selected it says some nodes are still filtered, this is not a regression and i think i'm just noticing as i'm looking for little things, but maybe a tooltip? I don't even know what they are.

File: app/api_topologies.go [hunk 2]
```diff
@@ -24,33 +24,69 @@ var (
...
-				{"hide", "System pods hidden", true, render.FilterSystem},
-			}},
+			Options: []APITopologyOptionGroup{
+				{
+					ID:      "system",
+					Default: "application",
+					Options: []APITopologyOption{
+						{"system", "System pods", render.FilterApplication},
+						{"application", "Application pods", render.FilterSystem},
+						{"both", "Both", render.FilterNoop},
```
Comment (tomwilkie) [inline app/api_topologies.go]: In https://github.com/weaveworks/scope/issues/1116#issuecomment-195478078, @rade seems to be implying that both should be in the middle.

Comment (rade) [inline app/api_topologies.go]: That very much depends on how this gets rendered.

Comment (tomwilkie) [inline app/api_topologies.go]: They get rendered in that order; @davkal should be able to provide some screenshots.

**Comment by davkal** [comment]
Current implementation:

screen shot 2016-03-24 at 14 11 28

*Both* in the middle:
screen shot 2016-03-24 at 14 12 31

**Comment by davkal** [comment]
@foot changed BG of unselected ones slightly and isolated topology caches, so nodes should overlap less PTAL

**Comment by foot** [comment]
Topos look good!

Not sure about the BG! Looks a bit heavy now, I'd remove that actually :P.

But LGTM either way. :+1:

**Comment by davkal** [comment]
Fixes #1116

**Pull request merged**